drv/transceivers-server/src/main.rs

```diff
@@ -272,9 +274,18 @@ impl ServerImpl {
                         self.thermal_models[i] =
                             self.decode_interface(port, interface)
                     }
+                    Err(FpgaError::ImplError(e)) => {
+                        ringbuf_entry!(Trace::GetInterfaceError(
+                            i,
+                            Reg::QSFP::PORT0_STATUS::Encoded::from_u8(e)
+                                .unwrap()
```

Comment:
I'm skeptical of this `unwrap()`.  If `FpgaError::ImplError` is _always_ intended to contain a valid `Encoded`, then we should make it contain that by construction (instead of `u8`); otherwise, we should probably be tolerant of a bad value here (logging something else to the ringbuf if decoding fails).

Reply:
Yeah, you are right. This is too aggressive. I'll handle the returned `Option<u8>`.